### ISSUE TITLE
[ADAP-849] [Bug] debug prints `private_key` and `private_key_passphrase` to stdout

### ISSUE THREAD

**Issue by louis-vines** [dbt-labs/dbt-snowflake#754]
### Is this a new bug in dbt-snowflake?

- [X] I believe this is a new bug in dbt-snowflake
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

I'm not sure this is technically a bug but not sure what else to put it as...

When calling the command `dbt debug`, the command prints my `private_key` and `private_key_passphrase` to stdout as part of the output. This is at odds to the behaviour of the `password` value, which is seems to be ommited from the output of `dbt debug`. If using private_key and private_key_passphrase for authentication they function in a similar way to a password so shouldn't they be sensored from the output in a similar fashion?

The current behaviour means I can't run `dbt debug` in a production environment as it will spurt my credentials into the logs.

### Expected Behavior

When running `dbt debug` with a profile configured to use private_key and private_key_passphrase, omit these keys from the output

### Steps To Reproduce

1. When configuring a profile with private_key and  private_key_passphrase
2. Try setting up a profiles.yml like so:

```
eunice:
  target: test

  outputs:
    test:
      type: snowflake
      account: MY_SNOWFLAKE_ACCOUT
      user: USER
      role: TRANSFORMER

      private_key: my_dummy_private_key
      private_key_passphrase: my_dummy_private_key_passphrase

      database: MY_DB
      schema: MY_SCHEMA
      warehouse: WAREHOUSE
```

3. run `dbt debug`
4. see the output:

```
14:57:29  Connection:
14:57:29    account: MY_SNOWFLAKE_ACCOUT
14:57:29    user: USER
14:57:29    database: MY_DB
14:57:29    warehouse: WAREHOUSE
14:57:29    role: TRANSFORMER
14:57:29    schema: MY_SCHEMA
14:57:29    authenticator: None
14:57:29    private_key: my_dummy_private_key
14:57:29    private_key_path: None
14:57:29    private_key_passphrase: my_dummy_private_key_passphrase
```

### Environment

```markdown
- OS: macos
- Python: 3.10.12
- dbt-core: 1.6.1
- dbt-snowflake: 1.6.0
```

**Comment by dataders** [comment]
@louis-vines great catch! you're right that this info certainly shouldn't be logged. the good news is that it is an easy fix.

`SnowflakeCredentials._connection_keys()` defines the connection profile attributes to be sent to stout by `debug`. Removing `private_key` and `private_key_passphrase` will resolve your issue. Additionally, I'm thinking that likely `token` should also be removed.

If you're interested, this would be a great opportunity for a community member to contribute a PR with the above fix -- I'd be glad to help your or anyone else out in doing so.

https://github.com/dbt-labs/dbt-snowflake/blob/43d12bc040069d65244637bde04576fc2cf65dfc/dbt/adapters/snowflake/connections.py#L106-L118

**Comment by louis-vines** [comment]
Hi @dataders I'm happy to contribute this improvement, sounds like it should be quite straight forward. I've put it on my todo list for the week...

**Comment by dataders** [comment]
@louis-vines sorry to lead you astray, we decided on Monday that represented a security issue and therefore prioritized shipping a fix that's now released in [dbt-snowflake 1.6.2](https://github.com/dbt-labs/dbt-snowflake/releases/tag/v1.6.2).

Thanks again for reporting, and if you would ever like to contribute to dbt-snowflake in the future the ["good_first_issue"](https://github.com/dbt-labs/dbt-snowflake/issues?q=is%3Aopen+is%3Aissue+label%3Agood_first_issue+) is a great place to start! Cheers
